src/python_testing/TC_EWATERHTR_2_1.py

```diff
@@ -0,0 +1,120 @@
+                     is_commissioning=True),
+            TestStep("2", "TH reads HeaterTypes attribute.",
+                     "DUT as Server replies with a WaterHeaterTypeBitmap (enum8) greater than 0x00 (at least one type supported), and less than 0x20 (no undefined types supported)."),
+            TestStep("3", "TH reads HeatDemand attribute.",
+                     "DUT as Server replies with a WaterHeaterDemandBitmap (enum8)."),
+            TestStep("4", "TH reads TankVolume attribute.",
+                     "DUT as Server replies with a uint16 value."),
+            TestStep("5", "TH reads EstimatedHeatRequired attribute.",
+                     "DUT as Server replies with an energy-mWh value."),
+            TestStep("6", "TH reads TankPercentage attribute.",
+                     "DUT as Server replies with a percent value."),
+            TestStep("7", "TH reads BoostState attribute.",
+                     "DUT as Server replies with a BoostStateEnum (enum8) value."),
+        ]
+
+        return steps
+
+    @async_test_body
+    async def test_TC_EWATERHTR_2_1(self):
+
+        em_supported = self.matter_test_config.global_test_params['PIXIT.EWATERHTR.EM']
+        tp_supported = self.matter_test_config.global_test_params['PIXIT.EWATERHTR.TP']
+
+        self.step("1")
+        # Commission DUT - already done
+
+        self.step("2")
+        heaterTypes = await self.read_whm_attribute_expect_success(attribute="HeaterTypes")
+        asserts.assert_greater(heaterTypes, 0,
+                               f"Unexpected HeaterTypes value - expected {heaterTypes} > 0")
+        asserts.assert_less_equal(heaterTypes, Clusters.WaterHeaterManagement.Bitmaps.WaterHeaterTypeBitmap.kOther,
+                                  f"Unexpected HeaterTypes value - expected {heaterTypes} <= WaterHeaterTypeBitmap.kOther")
+
+        self.step("3")
+        heatDemand = await self.read_whm_attribute_expect_success(attribute="HeatDemand")
+        asserts.assert_greater(heatDemand, 0,
+                               f"Unexpected HeatDemand value - expected {heatDemand} > 0")
+        asserts.assert_less_equal(heatDemand, Clusters.WaterHeaterManagement.Bitmaps.WaterHeaterDemandBitmap.kOther,
+                                  f"Unexpected HeatDemand value - expected {heatDemand} <= WaterHeaterDemandBitmap.kOther")
+
+        self.step("4")
+        if em_supported:
+            value = await self.read_whm_attribute_expect_success(attribute="TankVolume")
+            asserts.assert_greater_equal(value, 0, f"Unexpected TankVolume value - expected {value} >= 0")
+        else:
+            logging.info("Skipping step 4 as PIXIT.EWATERHTR.EM not supported")
+
+        self.step("5")
+        if em_supported:
+            value = await self.read_whm_attribute_expect_success(attribute="EstimatedHeatRequired")
+            asserts.assert_greater(value, 0, f"Unexpected EstimatedHeatRequired value - expected {value} > 0")
```

Comment:
Why are we checking the value is greater than 0? Does the `energy-mWh` not allow negatives? Is 0 impossible?

Reply:
Energy mWh is an int64 if I recall correctly.

0 technically is possible if it is already heated - so good spot - I think the spec PR I did just says it needs to be `min 0`.

https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/10028/files

So this should be changed to assert_greater_equal.